src/workflows/airqo_etl_utils/datautils.py

```
@@ -121,6 +102,43 @@
+            pd.DataFrame: A DataFrame formatted for BigQuery with required columns populated.
+
+        Raises:
+            KeyError: If the combination of data_type, device_category, and frequency is invalid.
+            Exception: For unexpected errors during column retrieval or data processing.
+        """
+        data.loc[:, "timestamp"] = pd.to_datetime(data["timestamp"])
+
+        try:
+            datasource = configuration.DataSource
+            cols = datasource.get(datatype).get(device_category).get(frequency)
+        except KeyError as e:
```

Ruff (0.8.2)

Local variable `e` is assigned to but never used

Remove assignment to unused variable `e`

(F841)